### ISSUE TITLE
 base: Can we use the file path as base?

### ISSUE THREAD

**Issue by ghpabs** [ludohenin/gulp-inline-ng2-template#10]
Given the following nested folder structure:
```
root/src/module1/component.ts
root/src/module1/component.html
```

I'd like to define the templateUrl as below:
```
// component.ts
templateUrl: 'component.html'
```

But I then need to set the `base` to `/src/module1` and this isn't very scalable.

---

I could easily fix it by changing the templateUrl to include the folder. But having the file path automatically set as base would be nicer.

```
// component.ts
templateUrl: '/module1/component.html'
```

Any thoughts?

**Comment by ludohenin** [comment]
I think we could, I had that in mind to avoid having to manage file paths.
I also need to check what Angular 2 team came with, but I beleive they have a solution for that in the beta.0 release.
For sure, we'll have to handle both case.
Do you have something to make a PR already ?

**Comment by ghpabs** [comment]
I was actually building a very similar module before I found yours. But we took a slight different approach to the problem so I don't think I can make a PR out of what I have here.

I couldn't find anything relevant in the beta.0 - hopefully there will be something like $templateCache was for 1.x.

**Comment by ludohenin** [comment]
What if I just add an option smthg like `relative: true` and use the file path as base ?

**Comment by ghpabs** [comment]
That sounds like the ideal thing to do. :+1:

**Comment by ludohenin** [comment]
e6ac647a81bb0579a1e5809db92781d4eff5baef solves this.

**Comment by ludohenin** [comment]
it's been published to npm as v0.0.10

**Comment by ghpabs** [comment]
Nice one!
